xo.py

```diff
@@ -0,0 +1,180 @@
+import random
+import string
+
+ALL_LETTERS = string.ascii_lowercase
+DEFAULT_BOARD_SIZE = 3
+
+
+def start_game(board_size=DEFAULT_BOARD_SIZE):
+
+    players = gather_players_names()
+
+    while True:
+        # Randomize play order.
+        random.shuffle(players)
+
+        print('First player to start is: ' + players[0])
+
+        main_game_logic(players, board_size)
+
+        start_another = __get_console_input('Another game? (Y to restart)', 1, 1)
+        if start_another.lower() != 'y':
+            break
+
+
+def gather_players_names():
+    player_1_name = __get_console_input('Player1, what\'s your name?', 3, 20)
+    player_2_name = __get_console_input('Player2, what\'s your name?', 3, 20)
+
+    return [player_1_name, player_2_name]
+
+
+def __get_console_input(message, min_length, max_length):
+    while True:
+        limit_message = f'{min_length} characters only' if min_length == max_length \
+            else f'{min_length}-{max_length} characters length'
+
+        console_input = input(f'{message} ({limit_message})\n')
+        if not (min_length <= len(console_input) <= max_length):
+            print(f'input length must be {min_length}-{max_length} characters! please try again')
+
+        else:
+            return console_input
+
+
+def main_game_logic(players, board_size):
+    game_moves = 0
+    max_moves = board_size * board_size
+    current_player = 0
+    board = [[None] * board_size for i in range(board_size)]
+    is_game_over = False
+    while game_moves < max_moves and not is_game_over:
+        print_board(board)
+        player_move(board, players, current_player)
+        is_game_over = check_board_win(board)
+
+        if not is_game_over:
+            game_moves += 1
+            current_player = (current_player + 1) % 2
+
+    print_board(board)
+
+    end_data = f'{players[current_player]} won the game' if is_game_over else 'It\'s a DRAW!'
+
+    print(f'GAME ENDED - {end_data}\n\n')
+
+
+def check_board_win(board):
+    """
+    This method receives the game board array and checks if there is a streak and game ended
+
+    :param board:
+        a two dimensional array representing the board status with None\X\O values
+
+    :return:
+        True if there is a win on the board (cross\vertical\horizontal)
+    """
+    board_size = len(board[0])
```

Comment:
Why not pass `boare_size` as a parameter?

Reply:
@Redundancy of data. if the board size can be inferred from the actual data, why pass another parameter?